decide/administration/views.py

```diff
@@ -0,0 +1,235 @@
+
+    def get(self, request, auth_id):
+        try:
+            query = Auth.objects.all().values().filter(id=auth_id).get()
+        except ObjectDoesNotExist:
+            return Response({}, status=HTTP_404_NOT_FOUND)
+        return Response(query, status=HTTP_200_OK)
+
+    def put(self, request, auth_id):
+        if not AuthSerializer(data=request.data).is_valid():
+            return Response({"result": "Auth object is not valid"}, status=HTTP_400_BAD_REQUEST)
+        else:
+            try:
+                auth = Auth.objects.all().filter(id=auth_id).get()
+            except ObjectDoesNotExist:
+                return Response({}, status=HTTP_404_NOT_FOUND)
+            for key, value in request.data.items():
+                setattr(auth, key, value)
+            auth.save()
+            return Response({}, status=HTTP_200_OK)
+
+    def delete(self, request, auth_id):
+        Auth.objects.all().filter(id=auth_id).delete()
+        return Response({}, status=HTTP_200_OK)
+
+
+class KeysAPI(APIView):
+    permission_classes = (IsAdminAPI,)
+
+    def get(self, request):
+        query = Key.objects.all().values()
+        return Response(query, status=HTTP_200_OK)
+
+    def post(self, request):
+        key = KeySerializer(data=request.data)
+        if not key.is_valid():
+            return Response({"result", "Key object is not valid"}, status=HTTP_400_BAD_REQUEST)
+        else:
+            key.save()
+            return Response({}, status=HTTP_200_OK)
+
+    def delete(self, request):
+        if request.data["idList"] is None:
+            Key.objects.all().delete()
+            return Response({}, status=HTTP_200_OK)
+        else:
+            ids = request.data["idList"]
+            is_valid(len(ids) > 0, 'The ids list can not be empty')
+            Key.objects.filter(id__in=ids).delete()
+            return Response({}, status=HTTP_200_OK)
+
+
+class KeyAPI(APIView):
+    permission_classes = (IsAdminAPI,)
+
+    def get(self, request, key_id):
+        try:
+            query = Key.objects.all().values().filter(id=key_id).get()
+        except ObjectDoesNotExist:
+            return Response({}, status=HTTP_404_NOT_FOUND)
+        return Response(query, status=HTTP_200_OK)
+
+    def put(self, request, key_id):
+        if not KeySerializer(data=request.data).is_valid():
+            return Response({"result": "User object is not valid"}, status=HTTP_400_BAD_REQUEST)
+        else:
+            try:
+                keym = Key.objects.all().filter(id=key_id).get()
+            except ObjectDoesNotExist:
+                return Response({}, status=HTTP_404_NOT_FOUND)
+            for key, value in request.data.items():
+                setattr(keym, key, value)
+            keym.save()
+            return Response({}, status=HTTP_200_OK)
+
+    def delete(self, request, key_id):
+        Key.objects.all().filter(id=key_id).delete()
+        return Response({}, status=HTTP_200_OK)
+
+
+class UsersAPI(APIView):
```

Comment:
Esto debería formar parte del modulo Authentication, no de Administration

Reply:
Puesto que aquí se hace uso del auth_token de django y se requiere del seteo de la cookie que el modulo de administración usa creo que la integración debería ser realizada a nivel de backend pero manteniendo los endpoints de autenticación particulares del modulo de administración.

Podríamos plantear el uso de una función semejante a la que django ofrece por defecto: `authenticate` que realice log in y devuelva un objeto similar, con el `auth_token`